File: Q4_Search.py
class Node:
    def __init__(self,data):
        self.data=data
        self.next=None
def search(h,key):
    while h:
        if h.data==key:
            return True
        h=h.next
    return False
head=Node(1)
head.next=Node(2)
print(search(head,2))

File: Q4_Search.c
#include <stdio.h>
#include <stdlib.h>
struct Node{int data;struct Node* next;};
int search(struct Node* h,int key){while(h){if(h->data==key)return 1;h=h->next;}return 0;}
int main(){struct Node* head=malloc(sizeof(struct Node));head->data=1;head->next=malloc(sizeof(struct Node));head->next->data=2;head->next->next=NULL;printf("%d",search(head,2));}